src/Microsoft.AspNetCore.Razor/Parser/TagHelpers/TagHelperDirectiveSpanVisitor.cs

```diff
@@ -87,26 +87,35 @@ public override void VisitSpan(Span span)
                 throw new ArgumentNullException(nameof(span));
             }
 
+            string directiveText = null;
             TagHelperDirectiveType directiveType;
-            if (span.ChunkGenerator is AddTagHelperChunkGenerator)
+
+            var addTagHelperChunkGenerator = span.ChunkGenerator as AddTagHelperChunkGenerator;
+            var removeTagHelperChunkGenerator = span.ChunkGenerator as RemoveTagHelperChunkGenerator;
+            var tagHelperPrefixChunkGenerator = span.ChunkGenerator as TagHelperPrefixDirectiveChunkGenerator;
+
+            if (addTagHelperChunkGenerator != null)
```

Comment:
Casting anyways, thought might as well just use the as syntax instead of `is` => `((AddTagHelperChunkGenerator)span.ChunkGenerator).LookupText`